Update Dashboard UI

stefanbitcr opened the pull request:
## 📝 Description

* Display On Chain Balance
* Display Credit Balance
* Display Debit Balance
* Show initials as avatar rather than stock images
* Show logged in Keycloak account as user
* Remove pages that aren't currently used
* Use a single quote page for all different statuses (deleted other quote listing pages as they are duplicates)
* Allow anon types for payee
* Remove mock data
* Refresh auth token
* Only allow keyset activation for accepted quotes

---

## ✅ Checklist

Please ensure the following tasks are completed before requesting a review:

- [ ] My code adheres to the style guidelines of this project.
- [x] I have run `npm run lint` or the equivalent linting command.
- [ ] I have added or updated tests (if applicable).
- [ ] My changes have been tested thoroughly in different browsers/resolutions (if applicable).
- [ ] I have updated the documentation (if applicable).
- [ ] I have checked that there are no console errors or warnings.
- [x] I have verified that the application builds without errors.
- [ ] I have tested responsiveness for mobile and desktop views (if applicable).

stefanbitcr committed:
Remove mocks

stefanbitcr committed:
Add debit and credit types

stefanbitcr committed:
Show balance

stefanbitcr committed:
Hide earnings and info from side bar

stefanbitcr committed:
Update payee data type

stefanbitcr committed:
Refresh keycloak

stefanbitcr committed:
Display Ecash balance

stefanbitcr committed:
Display anon payee

stefanbitcr committed:
Use generic quote page for different statusses

stefanbitcr committed:
Use initials as avatar

stefanbitcr committed:
Use correct avatar

stefanbitcr committed:
Update balance page

stefanbitcr committed:
Remove dev mode

stefanbitcr committed:
Remove dev mode

stefanbitcr committed:
Display keycloak user

stefanbitcr committed:
Display keycloak user

stefanbitcr committed:
Fetch on chain balance

stefanbitcr committed:
Remove comment

stefanbitcr committed:
Remove unnecessary import

stefanbitcr committed:
Only allow keyset activation for accepted quotes

cryptosalomao commented on src/components/nav/NavUser.tsx:
```diff
@@ -31,14 +34,19 @@ export function NavUser({
-                <AvatarImage src={user.avatar} alt={user.name} />
-                <AvatarFallback className="rounded-lg">{user.name}</AvatarFallback>
+                <div
+                  className="w-full h-full flex items-center justify-center text-white font-semibold text-sm"
```
You may use the custom background color directly with the `bg-` class utiltity to avoid creating extra variables and passing it down through properties, for example: `bg-[#f59e0b]`

```suggestion
                  className="w-full h-full flex items-center justify-center text-white font-semibold text-sm bg-[#f59e0b]"
```

cryptosalomao commented on src/pages/balances/BalancesPage.tsx:
```diff
@@ -120,86 +118,196 @@ export function OtherBalanceChart() {
 }
 
-function PageBody() {
-  const { data } = useSuspenseQuery({
-    queryKey: ["balances"],
-    queryFn: fetchBalances,
+function isErrorResponse<T>(res: { data?: T; error?: unknown }): res is { data: undefined; error: unknown } {
+  return res.error !== undefined
+}
```
Is there a special reason to stop using tanstack query here? If it's suspense mode, I understand that it may be tricky to handle multiple queries at once with it, but you could still use the `useQueries` hook to call multiple functions at once, this way you wouldn't have to handle all the individual state updates and errors, and could still access the status imperatively with `isLoading`, `isError`, `isSuccess && data``, etc.

For reference:
https://tanstack.com/query/latest/docs/framework/react/reference/useQueries

Reference

cryptosalomao commented on src/pages/balances/BalancesPage.tsx:
```diff
@@ -120,86 +118,196 @@ export function OtherBalanceChart() {
+  useEffect(() => {
+    void fetchAllBalances()
+
+    const interval = setInterval(() => void fetchAllBalances(), 30000)
```
All good here, just a suggestion to separate the decimals here for better readability xD

```suggestion
    const interval = setInterval(() => void fetchAllBalances(), 30_000)
```

cryptosalomao commented on src/pages/quotes/QuotePage.tsx:
```diff
@@ -1,24 +1,35 @@
+import { getDeterministicColor } from "@/utils/dev"
+
+// Utility functions for deterministic avatars
+function getInitials(name?: string): string {
+  if (!name) return "?"
+  const words = name.trim().split(/\s+/)
+  if (words.length === 1) {
+    return words[0].substring(0, 2).toUpperCase()
+  }
+  return (words[0][0] + words[words.length - 1][0]).toUpperCase()
+}
```
Maybe move this to it's own utils file? Not sure if it's an early optimization tho, please tell me. Also, we have this very same utility with a cleaner functional approach in our frontend repo:

https://github.com/BitcreditProtocol/E-Bill-frontend/blob/main/src/utils/strings.ts#L11

We have other util functions there too, so please feel free to use as many of those as you want to!

cryptosalomao commented:
Only a few things and quick improvements suggestions. The major thing is the use of tanstack query that has been reduced in here. It works with anything wrapped in a Promise, so you wouldn't have to handle all the individual statuses (fulfilled, pending, rejection, etc) manually.

I think it's worth the time to try and learn the basics of it to keep redundancy minimal:
https://tanstack.com/query/latest/docs/framework/react/guides/queries#query-basics

stefanbitcr commented:
@cryptosalomao Thank you for the review effort! I will take a look at tanstack, indeed my unfamiliarity with it might be overcomplicating things!

cryptosalomao commented:
> @cryptosalomao Thank you for the review effort! I will take a look at tanstack, indeed my unfamiliarity with it (and react) might be overcomplicating things!

You're welcome! I'm sure you'll find it easy after giving it a try. Let me know if you need extra help and want to jump on a call to address any questions you may have

stefanbitcr committed:
Move functional get Initials to utils
Stolen from https://github.com/BitcreditProtocol/E-Bill-frontend/blob/d0356f045b0d11a5cc094c698199aa9a04139d77/src/utils/strings.ts#L11

stefanbitcr committed:
Directly set color using bg-

stefanbitcr committed:
Use BalanceDisplay for consistency

stefanbitcr committed:
Minor changes

stefanbitcr committed:
Use tanstack useQueries and add timeout

stefanbitcr committed:
Remove is pending
